runtime/drivers/file/file.go

```diff
@@ -28,20 +61,54 @@ func (d driver) Open(dsn string, logger *zap.Logger) (drivers.Connection, error)
 		return nil, err
 	}
 
-	c := &connection{root: absPath}
+	c := &connection{
+		root:         absPath,
+		driverConfig: config,
+		driverName:   d.name,
+	}
 	if err := c.checkRoot(); err != nil {
 		return nil, err
 	}
 	return c, nil
 }
 
-func (d driver) Drop(dsn string, logger *zap.Logger) error {
+func (d driver) Drop(config map[string]any, logger *zap.Logger) error {
 	return drivers.ErrDropNotSupported
 }
 
+func (d driver) Spec() drivers.Spec {
+	return spec
+}
+
+func (d driver) HasAnonymousSourceAccess(ctx context.Context, src drivers.Source, logger *zap.Logger) (bool, error) {
+	return true, nil
+}
+
+type sourceProperties struct {
+	Path   string `mapstructure:"path"`
+	Format string `mapstructure:"format"`
+}
+
+func parseSourceProperties(props map[string]any) (*sourceProperties, error) {
+	conf := &sourceProperties{}
+	err := mapstructure.Decode(props, &conf)
```

Comment:
Not sure it's a problem here, but I have previously seen issues in some libraries when parsing into a `**` (it uses `&` twice). Other places you use `mapstructure` in this PR only takes one level of pointer. Maybe fix to be safe?